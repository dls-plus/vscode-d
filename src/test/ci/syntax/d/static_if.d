static if (/* hello world */ a is b && is(ushort32)) ushort32[] array;
static assert (/**/ is(typeof(true)));
void main() { if(is(typeof(x)) == 5) {} }