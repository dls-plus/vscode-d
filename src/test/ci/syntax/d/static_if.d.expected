[{"start":0,"end":11,"scope":"keyword.control.static-if.begin.d"},{"start":11,"end":13,"scope":"comment.block.begin.d"},{"start":13,"end":26,"scope":"comment.block.content.d"},{"start":26,"end":28,"scope":"comment.block.end.d"},{"start":28,"end":29,"scope":"source.d"},{"start":29,"end":30,"scope":"variable.d"},{"start":30,"end":31,"scope":"source.d"},{"start":31,"end":33,"scope":"keyword.operator.identity.d"},{"start":33,"end":34,"scope":"source.d"},{"start":34,"end":35,"scope":"variable.d"},{"start":35,"end":36,"scope":"source.d"},{"start":36,"end":38,"scope":"keyword.operator.logical.d"},{"start":38,"end":39,"scope":"source.d"},{"start":39,"end":42,"scope":"keyword.token.is.begin.d"},{"start":42,"end":50,"scope":"variable.d"},{"start":50,"end":51,"scope":"keyword.token.is.end.d"},{"start":51,"end":52,"scope":"keyword.control.static-if.end.d"},{"start":52,"end":53,"scope":"source.d"},{"start":53,"end":61,"scope":"variable.d"},{"start":61,"end":62,"scope":"source.d"},{"start":62,"end":63,"scope":"source.d"},{"start":63,"end":64,"scope":"source.d"},{"start":64,"end":69,"scope":"variable.d"},{"start":69,"end":71,"scope":"meta.statement.end.d"}]
[{"start":0,"end":15,"scope":"keyword.other.static-assert.begin.d"},{"start":15,"end":17,"scope":"comment.block.begin.d"},{"start":17,"end":19,"scope":"comment.block.end.d"},{"start":19,"end":20,"scope":"source.d"},{"start":20,"end":23,"scope":"keyword.token.is.begin.d"},{"start":23,"end":30,"scope":"keyword.token.typeof.d"},{"start":30,"end":34,"scope":"constant.language.boolean.true.d"},{"start":34,"end":35,"scope":"keyword.token.typeof.d"},{"start":35,"end":36,"scope":"keyword.token.is.end.d"},{"start":36,"end":37,"scope":"keyword.other.static-assert.end.d"},{"start":37,"end":39,"scope":"meta.statement.end.d"}]
[{"start":0,"end":4,"scope":"storage.type.basic-type.d"},{"start":4,"end":5,"scope":"source.d"},{"start":5,"end":9,"scope":"entity.name.function.d"},{"start":9,"end":10,"scope":"source.d"},{"start":10,"end":11,"scope":"source.d"},{"start":11,"end":14,"scope":"source.d"},{"start":14,"end":16,"scope":"keyword.control.if.d"},{"start":16,"end":17,"scope":"source.d"},{"start":17,"end":20,"scope":"keyword.token.is.begin.d"},{"start":20,"end":27,"scope":"keyword.token.typeof.d"},{"start":27,"end":28,"scope":"variable.d"},{"start":28,"end":29,"scope":"keyword.token.typeof.d"},{"start":29,"end":30,"scope":"keyword.token.is.end.d"},{"start":30,"end":31,"scope":"source.d"},{"start":31,"end":32,"scope":"keyword.operator.assign.d"},{"start":32,"end":33,"scope":"keyword.operator.assign.d"},{"start":33,"end":34,"scope":"source.d"},{"start":34,"end":35,"scope":"constant.numeric.integer.decimal.d"},{"start":35,"end":36,"scope":"source.d"},{"start":36,"end":42,"scope":"source.d"}]